web/src/components/pages/profile/info/info.tsx

```diff
@@ -177,6 +177,12 @@ function ProfilePage({
         addNotification(getString('profile-form-submit-saved'));
       },
     ]);
+    if (window.location.search.includes('first=1')) {
+      sessionStorage.setItem('first', 'true');
+      window.location.assign(
+        `${window.location.origin}/${locale}/dashboard/challenge?${window.location.search}`
+      );
```

Comment:
Nice! This might be easier by setting `window.location.pathname`, like so:

```
window.location.pathname = `/${locale}/dashboard/challenge`
```

It should hold onto the `?` parameters.